mongo_connector/doc_managers/neo4j_doc_manager.py

```diff
@@ -33,13 +31,20 @@ class DocManager(DocManagerBase):
   def __init__(self, url, auto_commit_interval=DEFAULT_COMMIT_INTERVAL,
                  unique_key='_id', chunk_size=DEFAULT_MAX_BULK, **kwargs):
     self.remote_graph = Graph(url, **kwargs.get('clientOptions', {}))
-    graph = Graph()
+    self.graph = Graph()
     self.auto_commit_interval = auto_commit_interval
     self.unique_key = unique_key
     self.chunk_size = chunk_size
     if self.auto_commit_interval not in [None, 0]:
       self.run_auto_commit()
-  
+    self._formatter = DefaultDocumentFormatter()
+    self.query_nodes = {}
+    self.doc_types = []
+
+  def apply_id_constraint(self, doc_type):
+    constraint = "CREATE CONSTRAINT ON (d:{doc_type}) ASSERT d.id IS UNIQUE".format(doc_type=doc_type)
```

Comment:
So, ids are unique acress collections. Which lead us to have this, for example
```
{"_id":{"$oid":"55caddc301f4f2aa0b7ea772"},"room":"Auditorium","session":{"abstract":"Spring emerged as a core open source project in early 2003 and evolved to a broad portfolio of open source projects up until 2015. This keynote reflects upon the journey so far, with a focus on the open source ecosystem that Spring lives within, including stories and anecdotes from the old days as well as from recent times. Not getting stuck in history, we’ll also look at the continuity of Spring’s philosophy and its immediate applicability to Java development challenges in 2015 and beyond.","title":"12 Years of Spring: An Open Source Journey"},"speaker":{"bio":"Juergen Hoeller is co-founder of the Spring Framework open source project and has been serving as the project lead and release manager for the core framework since 2003. Juergen is an experienced software architect and consultant with outstanding expertise in code organization, transaction management and enterprise messaging.","name":"Juergen Hoeller","picture":"http://www.springio.net/wp-content/uploads/2014/11/juergen_hoeller-220x220.jpeg","twitter":"https://twitter.com/springjuergen"},"timeslot":"Wed 29th, 09:30-10:30","topics":["keynote","spring"]}
```
There's a id referencing the collection
But three nodes should be created here, a:Document:[collection_name] //in this case, for example, talk, b:document:session, c:Document:speaker. All under the same Mongo ID. So the nodes of the collection will carry the same ID. Wouldnt it cause conflicts?